### ISSUE TITLE
Obj.magic() causes code execution to fail

### ISSUE THREAD

**Issue by jozanza** [Sketch-sh/sketch-sh#224]
I'm really loving Sketch 🎉, however, I ran into problems trying to use OCaml's `Queue` module -- the types all checked out, and code execution ran without issue, until I actually called `Queue.create()`.

After some digging, I realized that `Queue` uses some unsafe type coercion via `Obj.magic()` in a number of methods; `Queue.create()` is one of them.

Further testing revealed that `Obj.magic()` worked in cases involving `bool`, `int`, and `float`, but broke code execution when used to assign `string` or `record` type.

I have a Sketch that illustrates the problem in more detail:
https://sketch.sh/s/O5RAG6SC1rDRGzTu3d6GaU/

Right now, it seems like the only possible workaround is to roll my own `Queue` module. I'm still fairly new when it comes to reason/ocaml, but I'd love to help out if possible.

**Comment by thangngoc89** [comment]
Hello, thank you for using and loving Sketch.sh. `Queue` isn't part of the stdlib, though it's often shipped with the compiler. Currently Sketch only supports the stdlib (at OCaml 4.06). There are plans to support loading external packages. 

Regardings your problem, I think this is an upstream issue (with JSOO) or maybe how your Queue is implemented. I will have tho dig deeper to see what the actual issue is.

In the mean time, here is a `Queue` implementation you can use that works with Sketch. 

https://sketch.sh/s/YAdMoaIgTvVcdGGeOs1njq/

**Comment by Drup** [comment]
[Queue **is** part of the standard library](http://caml.inria.fr/pub/docs/manual-ocaml/libref/Queue.html).

**Comment by thangngoc89** [comment]
@Drup I'd only consider anythings that's linked with `-I +stdlib` to be the *real* stdlib. Anything else isn't. For ex: `Str`, `Unix`

**Comment by Drup** [comment]
Yes, and Queue is part of that. The content of the stdlib is shown [here](http://caml.inria.fr/pub/docs/manual-ocaml/stdlib.html).

**Comment by thangngoc89** [comment]
@Drup interesting. I wonder why sketch doesn't have Queue module

**Comment by jozanza** [comment]
Thanks for the alternate implimentation @thangngoc89!

FWIW, I've been using the [reasonml docs](https://reasonml.github.io/api/Queue.html) as a stdlib reference. When I realized `Queue` references broke my Sketch, I just copy-pasted the  [vendored ocaml dependency from bsb-native](https://github.com/bsansouci/bsb-native/blob/master/vendor/ocaml/stdlib/queue.ml), which appears to have been authored by INRIA back in 2002. That's where the `Obj.magic()` came from in the first place. I never even knew that was a thing prior 😆

**Comment by Drup** [comment]
FWIW, the [current version](https://github.com/ocaml/ocaml/blob/5ccf3c52dcbff5b5fc76b127ddb19aca90090b53/stdlib/queue.ml) of Queue doesn't use Obj.magic

**Comment by jozanza** [comment]
@Drup Thanks that's exactly what I needed! Not sure why `bsb-native` is using a weird out-of-date version 🤷‍♂
